src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -92,6 +93,21 @@ public HttpConnection(Uri url, HttpTransportType transportType, ILoggerFactory l
                 _httpClient = CreateHttpClient();
             }
 
+            // Set the X-Requested-With header on httpOptions for WebSockets
```

Comment:
Hmm, this feels kind of messy. What about setting this header manually inside the `WebSocketsTransport`. It keeps the transport specific logic closer to the implementation.

Reply:
Yeah, we were talking about that, not sure if we landed anywhere

Reply:
Just do it on the ClientWebSocketOptions and HttpClient directly